strawberry/chalice/views.py

```diff
@@ -0,0 +1,153 @@
+            accept_headers = headers.get("accept")
+        except (KeyError, TypeError):
+            return False
+
+        if accept_headers is None:
+            return False
+
+        if "text/html" in accept_headers:
+            return True
+
+        if "*/*" in accept_headers:
+            return True
+
+        return False
+
+    @staticmethod
+    def method_is_post(method: str) -> bool:
+        """
+        If the get method has been called this is True
+        Args:
+            method: A request verb such as GET, POST, PUT etc.
+
+        Returns:
+            True if the method is post
+        """
+        if method == "POST":
+            return True
+
+        return False
+
+    @staticmethod
+    def method_is_get(method: str) -> bool:
+        """
+        If the get method has been called this is True
+        Args:
+            method: A request verb such as GET, POST, PUT etc.
+
+        Returns:
+            True if the method is get
+        """
+        if method == "GET":
+            return True
+
+        return False
+
+    @staticmethod
+    def invalid_query_response() -> Response:
+        """
+        A response for malformed queries
+        Returns:
+        An errors response
+        """
+        return Response(
+            body={"errors": ["Provide a valid query / body to your call"]},
+            status_code=HTTPStatus.OK,
+        )
+
+    @staticmethod
+    def invalid_rest_verb_response() -> Response:
+        """
+        A response for calling the graphql endpoint with a non POST request
+        Returns:
+
+        """
+        return Response(
+            body={"errors": ["GraphQL queries must be of request type POST"]},
+            status_code=HTTPStatus.OK,
+        )
+
+    def execute_request(self, request: Request) -> Response:
+        """
+        Parse the request process it with strawberry and return a response
+        Args:
+            request: The chalice request this contains the headers and body
+
+        Returns:
+            A chalice response
+        """
+        if self.graphiql:
+            if self.has_html_been_asked_for(
+                headers=request.headers
+            ) and self.method_is_get(request.method):
+                graphiql_page: str = self.render_graphiql()
+                return Response(
+                    body=graphiql_page,
+                    headers={"content-type": "text/html"},
+                    status_code=200,
+                )
+
+        if not self.method_is_post(request.method):
+            return self.invalid_rest_verb_response()
+
+        request_data = request.json_body
+
+        if request_data is None:
+            return self.invalid_query_response()
+        try:
+            query = request_data["query"]
+            variables = request_data.get("variables")
+            operation_name = request_data.get("operationName")
+
+        except (KeyError, TypeError):
+            return self.invalid_query_response()
```

Comment:
When would these raise `TypeError`s?

Reply:
If people put weird stuff into their json requests. It can happen. I.E. saying it's application/json then putting a string in as the request (not json)

Reply:
Im pretty sure these `get`s won't raise TypeErrors, but `request.json_body` a few lines above could raise an error when parsing the raw body. (See Chalice source code: https://github.com/aws/chalice/blob/8df8b59aa759292ec065dae54a074731b11fa28e/chalice/app.py#L437)

Reply:
Have made the above changes

Reply:
Added the catch around the request.json_body. The TypeError will get raised if the payload is reported as application/json, but a string is provided I believe, so I've got a test case for that and left it in